notifier/notifier.go

```diff
@@ -383,25 +391,25 @@ func (n *Notifier) Stop() {
 
 // Describe implements prometheus.Collector.
 func (n *Notifier) Describe(ch chan<- *prometheus.Desc) {
```

Comment:
At least the way it is right now, with half of the metrics being registered with the passed-in `Registerer` and the other half being collected via the collector interface is not very nice. Choose either one or the other fully, I'd say.